src/src/com/microsoft/aad/adal/BrokerProxy.java

```diff
@@ -124,38 +127,83 @@ public boolean verifyUser(String username, String uniqueid) {
 
     @Override
     public boolean canUseLocalCache() {
-        boolean brokerSwitch = canSwitchToBroker();
+        boolean brokerSwitch;
+        try {
+            brokerSwitch = canSwitchToBroker();
+        } catch (final UsageAuthenticationException e) {
+            Logger.w(
+                    TAG,
+                    e.getMessage(),
+                    "", ADALError.DEVELOPER_BROKER_PERMISSIONS_MISSING);
+            brokerSwitch = false;
+        }
         if (!brokerSwitch) {
             Logger.v(TAG, "It does not use broker");
             return true;
         }
 
-        String packageName = mContext.getPackageName();
+        final String packageName = mContext.getPackageName();
         if (verifySignature(packageName)) {
             Logger.v(TAG, "Broker installer can use local cache");
             return true;
         }
 
         return false;
     }
-
+        
     /**
-     * App needs to give permission to AccountManager to use broker.
+     * To verify if App gives permissions to AccountManager to use broker.
+     * 
+     * @throws UsageAuthenticationException
+     * @return boolean If the GET_ACCOUNTS, MANAGE_ACCOUNTS, USE_CREDENTIALS permissions are granted in the Manifest.xml
      */
-    private boolean verifyManifestPermissions() {
-        PackageManager pm = mContext.getPackageManager();
-        boolean permission = PackageManager.PERMISSION_GRANTED == pm.checkPermission(
-                "android.permission.GET_ACCOUNTS", mContext.getPackageName())
-                && PackageManager.PERMISSION_GRANTED == pm.checkPermission(
-                        "android.permission.MANAGE_ACCOUNTS", mContext.getPackageName())
-                && PackageManager.PERMISSION_GRANTED == pm.checkPermission(
-                        "android.permission.USE_CREDENTIALS", mContext.getPackageName());
-        if (!permission) {
-            Logger.w(TAG, "Broker related permissions are missing for GET_ACCOUNTS, MANAGE_ACCOUNTS, USE_CREDENTIALS",
+    private boolean verifyManifestPermissions() throws UsageAuthenticationException {
+        final String methodName = ":verifyManifestPermissions";
+        final PackageManager packageManager = mContext.getPackageManager();
+        List<String> permerssionMissing = new ArrayList<String>();
```

Comment:
like I mentioned in your closed PR, I don't think you need to create a list for this.